correctness/correctness_test.go

```diff
@@ -11,155 +11,616 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the speci
 and limitations under the License.
 */
 
-// This package executes all types of PromQLs Prometheus may send to the Prometheus Connector, which will output the
-// responses for the PromQLs to expectedOutput.csv for manual verification. The intent of this test is to ensure the
-// Prometheus Connector can properly translate PromQLs to Amazon Timestream SQLs.
-//
-// Prior to running the tests in this file, ensure valid IAM credentials are specified in the basic auth section within
-// config/prometheus.yml.
+// This test suite validates the correctness of the Prometheus Connector using Mockmetheus
+// to execute remote-read and remote-write operations. It covers various scenarios by
+// executing queries against a locally hosted Connector connected to a Timestream database.
 package correctness
 
 import (
-	"bufio"
 	"context"
-	"encoding/csv"
-	"io"
+	"fmt"
+	"math/rand"
 	"os"
-	"path/filepath"
+	"strings"
 	"testing"
 	"time"
-	"timestream-prometheus-connector/integration"
-	"timestream-prometheus-connector/timestream"
+)
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+// Enable this flag when working with a fresh Timestream database.
+var freshTSDB = false
```

Comment:
Can we make this an actual flag that we can set in the command line for overriding, without needing to change any code. Also good to add a section to the README for flags.